emukit/bayesian_optimization/loops/bayesian_optimization_loop.py

```diff
@@ -55,11 +57,32 @@ def __init__(self, model: IModel, space: ParameterSpace, acquisition: Acquisitio
 
         super().__init__(candidate_point_calculator, model_updaters, loop_state)
 
-    def get_current_best(self) -> dict:
+    def get_results(self):
+        return BayesianOptimizationResults(self.loop_state)
+
+
+class BayesianOptimizationResults:
+    def __init__(self,loop_state: LoopState):
+
         """
-        Returns a dictionary with the current best found location (minimum) and its value.
+        Emukit class that takes as input the loop state and computes some results.
 
+        :param loop_state: The loop state it its current form. Currently it only contains X and Y.
         """
-        return {'minimum': self.loop_state.X[np.argmin(self.loop_state.Y),None], 'value':np.min(self.loop_state.Y)}
 
+        self.minimum_location = loop_state.X[np.argmin(loop_state.Y),:]
+        self.minimum_value = np.min(loop_state.Y)
+        self.best_found_value_per_iteration = self._current_best_vector(loop_state.Y)
+
+    def _current_best_vector(self,Y) -> np.array:
+        '''
+        Returns a one-dimensional numpy array whose ith element us the minimum of the components 0:i of Y.
+
+        :param Y: the numpy array with the current best found value per iteration,
+        '''
+
+        Y_best = Y.flatten().copy()
```

Comment:
`return np.minimum.accumulate(Y).flatten()` does this method in one line

Reply:
thanks! done!